### PR TITLE
Added docs on python based backends

### PR DESCRIPTION
**Pull request by oandreeva-nv**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,6 +63,8 @@ Major features include:
   for stateful models
 - Provides [Backend API](https://github.com/triton-inference-server/backend) that
   allows adding custom backends and pre/post processing operations
+- Supports model architecture agnostic python scripts as custom
+  python based backends.
```
Comment (rmccorm4) [inline README.md:67]: Can we expand on it? Link to more detailed docs in python backend? Maybe we can also point to vLLM backend as an example somewhere.

Comment (oandreeva-nv) [reply README.md:67]: More docs in backend repo are comming. I am fixing pre-commit hooks:  https://github.com/triton-inference-server/backend/pull/88

File: README.md [hunk 2]
```diff
@@ -63,6 +63,8 @@ Major features include:
   for stateful models
 - Provides [Backend API](https://github.com/triton-inference-server/backend) that
   allows adding custom backends and pre/post processing operations
+- Supports model architecture agnostic python scripts as custom
```
Comment (nnshah1) [inline README.md:66]:
```suggestion
- Supports writing custom backends in python.
```

File: README.md [hunk 3]
```diff
@@ -63,6 +63,8 @@ Major features include:
   for stateful models
 - Provides [Backend API](https://github.com/triton-inference-server/backend) that
   allows adding custom backends and pre/post processing operations
+- Supports writing custom backends in python.
+  python based backends.
```
Comment (oandreeva-nv) [inline README.md:67]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -63,6 +63,7 @@ Major features include:
   for stateful models
 - Provides [Backend API](https://github.com/triton-inference-server/backend) that
   allows adding custom backends and pre/post processing operations
+- Supports writing custom backends in python.
```
Comment (rmccorm4) [inline README.md:66]: Assuming we have some high level docs for python-based backends somewhere, let's link it here. Similar to contents of this [comment](https://github.com/triton-inference-server/backend/pull/88#discussion_r1358893542).

**Review by tanmayv25 — APPROVED** [review]

**Pull request merged**